types/react/index.d.ts

```diff
@@ -2114,7 +2114,30 @@ declare namespace React {
         size?: number;
         src?: string;
         step?: number | string;
-        type?: string;
+        type?:
+            | 'button'
+            | 'checkbox'
+            | 'color'
+            | 'date'
+            | 'datetime-local'
+            | 'datetime'
```

Comment:
may be best to leave this one out.  after all, someone can always add it back in if they really need it.
```suggestion
```